fmt/med: implemented reading song title

logiclrd opened the pull request:
<img width="770" height="566" alt="image" src="https://github.com/user-attachments/assets/f1fb50b5-a769-40fc-94f1-7f239f8e5a1b" />

[Aliens_Par-Tay_Too2.zip](https://github.com/user-attachments/files/21824471/Aliens_Par-Tay_Too2.zip)

[MMD_FileFormat.txt](https://github.com/user-attachments/files/21824375/MMD_FileFormat.txt)

logiclrd committed:
fmt/med: implemented reading song title

mrpapersonic commented on fmt/med.c:
```diff
@@ -25,17 +25,98 @@
 #include "mem.h"
 
+// the most characters we will try to read for the song name.
+// we only care about 26 characters of song name, right??
```
what I do in the info readers is read as much as allowed and str_dup it into a buffer (or mem_alloc a buffer and read into it, both work but allocing first needs extra stuff to account for read fails)

if we're going to define an upper bounds for this it ought to be in `dmoz.h` where all of the readers can share it

logiclrd replied:
That's fair. Better to have the generic `fmt_read_info` function not apply any conventions; they're out of scope.

mrpapersonic commented on fmt/med.c:
```diff
@@ -25,17 +25,98 @@
+
+struct MMD0
+{
+	/* 0 */  char     id[4];
```
we probably shouldn't be loading a whole bunch of stuff we don't use if we don't have to. Just seek to the expdata pointer, read it in, and seek to it. Same with the name pointer in the exp structure.

(I don't see any major padding issues here, but it's good practice to NOT assume that the way the structure is laid out in memory will be the exact same as the disk format. In the past we used `#pragma pack(1)`, but this isn't very portable to oddball compilers, so I dumped all of that out.)

logiclrd replied:
Okay, that's fair. The flipside is that there's the outside chance that a full OctaMed loader might be implemented, and having the full header types could be useful. Maybe I'll leave the struct definition there but only read the fields needed at this point.

mrpapersonic replied:
yep, that's fine. That's what we do in many of the other loaders anyway (loading in each element one by one)

mrpapersonic commented on fmt/med.c:
```diff
@@ -25,17 +25,98 @@
+	struct MMD0 hdr;
+	struct MMD0exp exp;
+
+	int length = slurp_length(fp);
```
`int` -> `uint64_t`

mrpapersonic commented on fmt/med.c:
```diff
@@ -25,17 +25,98 @@
+	 || memcmp(hdr.id, MMD0_ID, 4))
+		return 0;
+
+	int exp_struct_ptr = bswapBE32(hdr.expdata_ptr);
```
`int` -> `uint32_t`

mrpapersonic commented on fmt/med.c:
```diff
@@ -25,17 +25,98 @@
+		return 0;
+
+	int name_ptr = bswapBE32(exp.songname_ptr);
+	int name_len = bswapBE32(exp.songnamelen);
```
`int` -> `uint32_t`

logiclrd committed:
fmt/med: PR feedback
Updated fmt_med_read_info to not truncate the name; that's the caller's job.
Reworked fmt_med_read_info to read only the fields needed. Left the structure definitions on the offchance that some future programmer wants to do a full OctaMed loader.
Fixed up data types in fmt_med_read_info.
